### PR TITLE
Script to update resource type versions in all Concourse pipelines

### PR DESCRIPTION
**Pull request by xoen**
Added a `update-pipeline-resource-type` script (see README).

This script is a first stab at the problem of updating existing pipelines when we tweak some of our custom resources.

Run it will patch the tag of the give resource type name with the desired version.

As part of this I've also spelled out the steps required to properly release a new version of this `github-org-resource` in the README.

### Background
Concourse pipeline can use custom resources (think of them like custom steps) by adding them to the `resource_types` section of the pipeline which will point to a Docker image/tag, see very bare example below (probably not working):

```yaml
---
jobs:
- name: deploy
  plan:
  - aggregate:
    - put: webapp-auth0-client
resources:
- name: webapp-auth0-client
  type: auth0-client
resource_types:
- name: auth0-client
  type: docker-image
  source:
    repository: quay.io/mojanalytics/concourse-auth0-resource
    tag: v2.0.2
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,6 +74,99 @@ jobs:
     trigger: true
 ```
 
+## Release process
+Change this resource is not enough to release it.
```
Comment (ntoll) [inline README.md]: Change to "Changing this resource..."

File: README.md [hunk 2]
```diff
@@ -74,6 +74,99 @@ jobs:
...
+A new release tag needs to be created and Concourse
+needs to be instructed to use it.
+
+Here the steps:
```
Comment (ntoll) [inline README.md]: "Here are the steps:"

File: README.md [hunk 3]
```diff
@@ -74,6 +74,99 @@ jobs:
...
+                                  Concourse pipeline
+
+  -i, --interactive               Run fly set-pipeline interactively
+  --help                          Show this message and exit.
+```
+
+Let's say for example that you want to update all the
+pipelines to use [`concourse-auth0-resource` `v2.0.2`](https://github.com/ministryofjustice/analytics-platform-concourse-auth0-client-resource/releases/tag/v2.0.2).
+
+First you need to find out how this resource is called
```
Comment (ntoll) [inline README.md]: how -> what

File: README.md [hunk 4]
```diff
@@ -74,6 +74,99 @@ jobs:
...
+in the pipeline, in this case if you check the `webapp`
+pipeline definition, in the `resource_types` section
+you'll see that this resource is instanciated with the
```
Comment (ntoll) [inline README.md]: instantiated

**Review by ntoll — APPROVED** [review]
There's a couple of grammatical errors in the README (see comments), but other than that it LGTM. :+1: :shipit: 

Like I said, I think we should try to automate all the things as much as possible and this is a great start. Thanks for getting things going. :rocket:

**Comment by xoen** [comment]
Thanks @ntoll 😬 : https://github.com/ministryofjustice/analytics-platform-concourse-github-org-resource/pull/44/commits/5910a1ea23de9e5bc2a015ffb3768bd442576905

**Comment by ntoll** [comment]
@xoen NP. :+1:

**Pull request merged**